ResourceUnavailable after calling some APIs containing write operations

Hi,

I am using py-trello for a project.  One of the dependency 'oauth2' was missing from the PyPI version of the package, which is very old (0.6.0).  I was able to figure the oauth2.Consumer class was present in later release of the package by using the latest [simplegeo/python-oauth2](https://github.com/simplegeo/python-oauth2).  However, the program seems to behave very weird.  I kept getting HTTP 400 randomly.  I was able to have some lucks with py-trello until making a call to anything contains write operation like below.  

The token requested contains read and write access.  Expiration was set to never.

I tested the same operations using _requests_.  It worked fine without issues.

``` python
client = TrelloClient(api_key=TRELLO_API_KEY, api_secret=TRELLO_API_SECRET,
                              token=token, token_secret=token_secret)

boards = client.list_boards()
b = boards[0]
print b
lists = b.all_lists()
try:
    for l in lists:
        l.add_card("Hello")
except:
    print "Something wrong"
print b.open_lists()
```

Here is the stack trace 

``` stacktrace
Traceback (most recent call last):
  File "test_py_trello.py", line 20, in <module>
    print b.open_lists()
  File "/Users/kennethfeng/anaconda/lib/python2.7/site-packages/trello/__init__.py", line 283, in open_lists
    return self.get_lists('open')
  File "/Users/kennethfeng/anaconda/lib/python2.7/site-packages/trello/__init__.py", line 293, in get_lists
    query_params = {'cards': 'none', 'filter': list_filter})
  File "/Users/kennethfeng/anaconda/lib/python2.7/site-packages/trello/__init__.py", line 177, in fetch_json
    raise ResourceUnavailable(url, response)

trello.ResourceUnavailable: Resource unavailable: https://api.trello.com/1/boards/*******************/lists?key=****************************&token=********************************&cards=none&filter=open (HTTP status: 400)
```

I forked the library and ran `nosetests -v test/`.  It seems to confirm my observation.

``` python
test01_list_boards (test_trello.TrelloClientTestCase) ... ok
test10_board_attrs (test_trello.TrelloClientTestCase) ... ok
test20_board_all_lists (test_trello.TrelloClientTestCase) ... ok
test21_board_open_lists (test_trello.TrelloClientTestCase) ... ok
test22_board_closed_lists (test_trello.TrelloClientTestCase) ... ok
test30_list_attrs (test_trello.TrelloClientTestCase) ... ok
test40_list_cards (test_trello.TrelloClientTestCase) ... ok
test50_add_card (test_trello.TrelloClientTestCase) ... FAIL
test51_add_card (test_trello.TrelloClientTestCase) ... FAIL

======================================================================
FAIL: test50_add_card (test_trello.TrelloClientTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kennethfeng/dev/repos/py-trello/test/test_trello.py", line 90, in test50_add_card
    self.fail("Caught Exception adding card")
AssertionError: Caught Exception adding card
-------------------- >> begin captured stdout << ---------------------
can't set attribute

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: test51_add_card (test_trello.TrelloClientTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kennethfeng/dev/repos/py-trello/test/test_trello.py", line 116, in test51_add_card
    self.fail("Caught Exception adding card")
AssertionError: Caught Exception adding card
-------------------- >> begin captured stdout << ---------------------
can't set attribute

--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 9 tests in 2.976s

FAILED (failures=2)
```

If you could quickly pin point the issue, that's great.  

If not, that's still fine.  It seems like it's better to migrate to requests from httplib2 and to rauth from oauth2.  If you agree, I will send you pull requests after I am done with the migration.

Kenneth

--- a/test/test_trello.py
+++ b/test/test_trello.py
@@ -1,6 +1,7 @@
 from trello import TrelloClient
 import unittest
 import os
+import datetime
 
 class TrelloClientTestCase(unittest.TestCase):
 
@@ -121,6 +122,37 @@ class TrelloClientTestCase(unittest.TestCase):
 				self.assertEquals(card.description, description)
 				self.assertIsNotNone(card.closed, msg="closed not provided")
 				self.assertIsNotNone(card.url, msg="url not provided")
+				break
+			break
+		if not card:
+			self.fail("No card created")
+
+	def test52_add_card_set_due(self):
+		boards = self._trello.list_boards()
+		board_id = None
+		for b in boards:
+			if b.name != os.environ['TRELLO_TEST_BOARD_NAME']:
+				continue
+
+			for l in b.open_lists():
+				try:
+					name = "Testing from Python"
+					description = "Description goes here"
+					card = l.add_card(name, description)
+				except Exception as e:
+					print str(e)
+					self.fail("Caught Exception adding card")
+
+                                # Set the due date to be 3 days from now
+                                today = datetime.datetime.today()
+                                day_detla = datetime.timedelta(3)
+                                due_date = today + day_detla #
+                                card.set_due(due_date)
+                                expected_due_date = card.due
+                                # Refresh the due date from cloud
+                                card.fetch()
+                                actual_due_date = card.due[:10]
+                                self.assertEquals(expected_due_date, actual_due_date)
 				break
 			break
 		if not card:

--- a/trello/__init__.py
+++ b/trello/__init__.py
@@ -519,7 +519,8 @@ class Card(object):
         self.idBoard = json_obj['idBoard']
         self.labels = json_obj['labels']
         self.badges = json_obj['badges']
-        self.due = json_obj['due']
+        # For consistency, due date is in YYYY-MM-DD format
+        self.due = json_obj.get('due', '')[:10]
         self.checked = json_obj['checkItemStates']
 
         self.checklists = []

